### PR TITLE
Fix gradient checkpointing in GaudiGemmaDecoderLayer + align forward() signature with HF 4.55

### PR DESCRIPTION
**Pull request by gplutop7**
This PR fixes a hidden_states=None failure in GaudiGemmaDecoderLayer when gradient checkpointing is enabled in distributed training.

Key changes

- Remove non-standard forward arguments (token_idx, reuse_cache, attn_softmax_bf16, use_flash_attention, etc.) from the method signature.
- Move all Gaudi-specific parameters to **kwargs and retrieve them via kwargs.get(...) (HF-compatible behavior).
- Add robust reconstruction logic for hidden_states when positional args are stripped by checkpointing wrappers.
- Ensure compatibility with Transformers 4.55 forward dispatching, PEFT, DDP, and Accelerate.
- Fully restore training stability on Gaudi with gradient checkpointing enabled

### CODE REVIEW

File: optimum/habana/transformers/models/gemma/modeling_gemma.py [hunk 1]
```diff
@@ -450,68 +460,84 @@ def update_sincos_cache(self, seq_len):
...
         - add new args attn_softmax_bf16
         """
+        token_idx = kwargs.get("token_idx", None)
+        attn_softmax_bf16 = kwargs.get("attn_softmax_bf16", False)
+        reuse_cache = kwargs.get("reuse_cache", False)
+        use_flash_attention = kwargs.get("use_flash_attention", False)
+        flash_attention_recompute = kwargs.get("flash_attention_recompute", False)
+        flash_attention_causal_mask = kwargs.get("flash_attention_causal_mask", False)
+        cache_idx = kwargs.get("cache_idx", None)
+        position_embeddings = kwargs.get("position_embeddings", None)
```
Comment (regisss) [inline optimum/habana/transformers/models/gemma/modeling_gemma.py:524]: Hmm weird we don't need to do that for other models no?

Comment (gplutop7) [reply optimum/habana/transformers/models/gemma/modeling_gemma.py:524]: Gradient checkpointing in Transformers 4.55 may call layers without positional args, passing only **kwargs (due to PyTorch CheckpointFunction + HF wrappers). We observed this in Gemma, hence the need to read these params from kwargs.
